### DISCUSSION TITLE
Creating unpaid issued invoices

### DISCUSSION THREAD

**Discussion by slipperyfrank** [fattureincloud/api#257] [Q&A] [answered]
I have been trying to create an issued document for an invoice that is yet to be paid but I keep receiving the error message "Il totale dei pagamenti non corrisponde al totale da pagare".

I could not find info suggesting unpaid invoices should be treated differently both in the API reference or developers guides, so all I did was setting the payments_list.status as "not_paid", what could I be missing?

```
{
   "data":{
      "entity":{
         "id":XXX,
         "name":"XXX",
         "first_name":"",
         "last_name":"",
         "vat_number":"XXX",
         "tax_code":"",
         "address_street":"XXX",
         "address_postal_code":"XXX",
         "address_city":"XXX",
         "address_province":"XX",
         "country":"Italia",
         "certified_email":"",
         "ei_code":"KRRH6B9"
      },
      "type":"invoice",
      "number":"289",
      "numeration":"\/W",
      "date":"2023-01-29",
      "subject":"XXX",
      "e_invoice":true,
      "ei_data":{
         "payment_method":"MP05"
      },
      "items_list":[
         {
            "description":"XXX",
            "net_price":"568.80"
         }
      ],
      "payments_list":[
         {
            "due_date":"2023-03-31",
            "status":"not_paid"
         }
      ],
      "amount_net":"568.80",
      "amount_vat":"125.14",
      "next_due_date":"2023-03-31"
   }
}
```

**Comment by valmoz** [comment]
Hi @slipperyfrank 
I suggest you check this guide: [Invoice Totals Guide](https://developers.fattureincloud.it/docs/guides/invoice-totals)

You can also find other discussions in this community about the same issue.

Thanks
Mauro

**Reply by slipperyfrank** [reply]
Hi @valmoz 
thanks, as I said I have checked the guides, particularly that one, and have also read the other discussions.

This specific example is for an invoice that has not been paid yet, so there is no "totale di pagamenti" in this case, that is why I mark the payments_list.status as "not_paid".

**Comment by emajo** [comment, marked as answer]
Hi @slipperyfrank, even if the invoice is not paid you must set the `payments_list` field to match the total `items_list` amount, so you must set at least a payment tranche with the amount even if the status is `not_paid`.

Thanks,
Emanuele.

**Reply by slipperyfrank** [reply]
Thanks @emajo, your answer was the one I found helpful in solving this problem.

I did try all combinations of null fields, payment statuses, payment terms, not setting any elements in the payments list etc. but to be honest the logic of indicating the amount of a payment that has not been made escaped me.

May I suggest this could be worded more explicitly in the guide to invoice totals? The guide opens with

> "IF you are setting the payments list of your invoice [...]"

which seemed to suggest that the payments_list field is not required (together with it not being marked with * in the API reference schema), that initially made sense to me as if there is no payment the payments list would be empty, though that is clearly not the case now.

Also the [guide to using the API reference](https://developers.fattureincloud.it/docs/api-reference/) states that "The fields marked with * are mandatory" but the schema detailed in the [API reference](https://developers.fattureincloud.it/api-reference/#post-/c/-company_id-/issued_documents) seems to miss the * for a number of fields which are indeed required, one of them being the payments_list and the payments_list[].amount (number┃null).

Maybe this is just me being pedantic, but I hope this would help fellow developers not having to figure it out by trial and error by having a strong unambiguous reference.

Thanks,
James
